muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/transcriptofrecords/rest/TranscriptofRecordsRESTService.java

```diff
@@ -224,14 +231,30 @@ public Response getVops(@PathParam("IDENTIFIER") String studentIdentifierString)
                 if (mandatority == Mandatority.MANDATORY) {
                   numMandatoryCourses++;
                 }
+
+                SchoolDataIdentifier gradingScaleIdentifier = workspaceAssessment.getGradingScaleIdentifier();
+                if (gradingScaleIdentifier == null) {
+                  break;
+                }
+                SchoolDataIdentifier gradeIdentifier = workspaceAssessment.getGradeIdentifier();
+                if (gradeIdentifier == null) {
+                  break;
+                }
+                GradingScale gradingScale = gradingController.findGradingScale(gradingScaleIdentifier);
```

Comment:
These two will most likely cause a performance issue by queries to Pyramus. Anything to optimize this with?